aptos/contracts/sources/state.move

```diff
@@ -132,75 +130,61 @@ module wormhole::state {
         );
     }
 
-    public(friend) fun update_guardian_set_index(newIndex: U32) acquires WormholeState {
+    public(friend) fun update_guardian_set_index(new_index: U32) acquires WormholeState {
+        let state = borrow_global_mut<WormholeState>(@wormhole);
+        state.guardian_set_index= new_index;
+    }
+
+    public fun get_guardian_set(index: U32): GuardianSet acquires WormholeState {
         let state = borrow_global_mut<WormholeState>(@wormhole);
-        state.guardian_set_index= newIndex;
+        *table::borrow<u64, GuardianSet>(&mut state.guardian_sets, u32::to_u64(index))
     }
 
-    public(friend) fun expire_guardian_set(_index: u64) acquires WormholeState {
-        let _state = borrow_global_mut<WormholeState>(@wormhole);
-        //TODO: expire guardian set, when we can index into guardian_sets with state.guardian_set_index(a U32)
-        //let guardian_set = table::borrow_mut<u64, GuardianSet>(&mut state.guardian_sets, state.guardian_set_index);
-        //Structs::expire_guardian_set(guardian_set);
+    public(friend) fun expire_guardian_set(index: U32) acquires WormholeState {
+        let state = borrow_global_mut<WormholeState>(@wormhole);
+        let guardian_set: &mut GuardianSet = table::borrow_mut<u64, GuardianSet>(&mut state.guardian_sets, u32::to_u64(index));
+        let expiry = state.guardian_set_expiry;
+        structs::expire_guardian_set(guardian_set, expiry);
     }
 
     public(friend) fun store_guardian_set(set: GuardianSet) acquires WormholeState {
         let state = borrow_global_mut<WormholeState>(@wormhole);
-        //TODO: store guardian set under index (U32)
-        //TODO: ensure that the guardian set index is incremental (maybe in the
-        // guardian_set_upgrade function)
-        let index: u64 = u32::to_u64(structs::get_guardian_set_index(set));
+        let index: u64 = u32::to_u64(structs::get_guardian_set_index(&set));
         table::add(&mut state.guardian_sets, index, set);
     }
 
+    public fun guardian_set_is_active(guardian_set: &GuardianSet): bool acquires WormholeState {
+        let index = structs::get_guardian_set_index(guardian_set);
+        let current_index = get_current_guardian_set_index();
+        let now = timestamp::now_seconds();
+
+        index == current_index ||
+            u32::to_u64(structs::get_guardian_set_expiry(guardian_set)) > now
+    }
+
     public(friend) fun set_governance_action_consumed(hash: vector<u8>) acquires WormholeState {
         let state = borrow_global_mut<WormholeState>(@wormhole);
         table::add(&mut state.consumed_governance_actions, hash, true);
```

Comment:
are you relying on `table::add` to error out if the hash has already been consumed? (since it would already be in the table)

Reply:
yes, that's right

Reply:
I've added unit tests for this to make sure it works as expected